src/lib/y2partitioner/dialogs/main.rb

```diff
@@ -104,6 +106,63 @@ def run(system, initial)
       def hostname
         @hostname ||= Yast::Hostname.CurrentHostname
       end
+
+      # Checks whether the dialog should be rendered again
+      #
+      # @return [Boolean]
+      def continue_running?(result)
+        if result == :redraw
+          true
+        elsif result == :abort && Yast::Mode.installation
+          !Yast::Popup.ConfirmAbort(:painless)
+        else
+          false
+        end
+      end
+
+      # Whether the user has accepted the Partitioner warning
+      #
+      # @note It returns true if the partitioner warning is not shown.
+      #
+      # @return [Boolean]
+      def accepted_partitioner_warning?
+        return true unless show_partitioner_warning?
+
+        partitioner_warning == :yes
+      end
+
+      # Popup to alert the user about the usage of the Partitioner
+      #
+      # @return [Symbol] user's answer (:yes, :no)
+      def partitioner_warning
+        # Warning popup about using the expert partitioner
+        message = _(
+          "This is for experts only.\n" \
+          "You might lose support if you use this!\n\n" \
+          "Please refer to the manual to make sure your custom\n" \
+          "partitioning meets the requirements of this product."
+        )
+
+        buttons = { yes: Yast::Label.ContinueButton, no: Yast::Label.CancelButton }
+
+        Yast2::Popup.show(message, headline: :warning, buttons: buttons, focus: :no)
```

Comment:
You can use `buttons: :continue_cancel`, so you don't need to construct your own `buttons` hash